src/Test/Tasty/Ingredients/Rerun.hs

```diff
@@ -200,8 +204,13 @@ rerunningTests :: [Tasty.Ingredient] -> Tasty.Ingredient
 rerunningTests ingredients =
   Tasty.TestManager (rerunOptions ++ Tasty.ingredientsOptions ingredients) $
     \options testTree -> Just $ do
-      let RerunLogFile stateFile = Tasty.lookupOption options
-          (UpdateLog updateLog, AllOnSuccess allOnSuccess, FilterOption filter)
+      stateFile <- case Tasty.lookupOption options of
+        DefaultRerunLogFile -> do
+            executableName <- takeBaseName <$> getExecutablePath
```

Comment:
Shall we use https://hackage.haskell.org/package/base-4.21.0.0/docs/System-Environment.html#v:executablePath instead? I'm fine with dropping support for older GHCs (or feel free to wrap it in CPP).

Reply:
I changed the PR to use that function instead.
If there is no reliable way to determine the executable path (or the executable got deleted while running the test suite) then we fall back to `.tasty-rerun-log`.